migrations/versions/0347_modify_table_and_function.py

```diff
@@ -0,0 +1,72 @@
+"""
+Revision ID: 0347_modify_table_and_function
+Revises: 0346_create_va_profile_functions
+Create Date: 2022-05-26 23:33:03.853609
+"""
+
+import sqlalchemy as sa
+from alembic import op
+from alembic_utils.pg_function import PGFunction
+
+revision = '0347_modify_table_and_function'
+down_revision = '0346_create_va_profile_functions'
+
+
+def upgrade():
+    public_va_profile_opt_in = PGFunction(
+        schema="public",
+        signature="va_profile_opt_in(_mpi_icn character varying, _va_profile_id integer, _communication_item_id integer, _communication_channel_name character varying)",
+        definition='returns void\n LANGUAGE sql\nAS $function$\n    INSERT INTO va_profile_local_cache (mpi_icn, va_profile_id, communication_item_id, communication_channel_name)\n    VALUES (_mpi_icn, _va_profile_id, _communication_item_id, _communication_channel_name)\n    ON CONFLICT DO NOTHING;\n$function$'
+    )
+    op.drop_entity(public_va_profile_opt_in)
+
+    public_va_profile_opt_out = PGFunction(
+        schema="public",
+        signature="va_profile_opt_out(_va_profile_id integer, _communication_item_id integer, _communication_channel_name character varying)",
+        definition='returns void\n LANGUAGE sql\nAS $function$\n    DELETE FROM va_profile_local_cache\n    WHERE va_profile_id = _va_profile_id AND communication_item_id = _communication_item_id AND communication_channel_name = _communication_channel_name;\n$function$'
+    )
+    op.drop_entity(public_va_profile_opt_out)
+
+    op.add_column('va_profile_local_cache', sa.Column('allowed', sa.Boolean(), nullable=False))
+    op.add_column('va_profile_local_cache', sa.Column('communication_channel_id', sa.Integer(), nullable=False))
+    op.add_column('va_profile_local_cache', sa.Column('source_datetime', sa.DateTime(), nullable=True))
```

Comment:
Why was this allowed to be nullable?

Reply:
@maham360 Did we discuss this?  I want to say there was a reason.

Reply:
Yes we did discuss this and we decided it should be True, I think because if we compare the `SourceDate` to `NULL` that means we received data and decided not to act on it